gluten-core/src/main/scala/io/glutenproject/execution/BasicPhysicalOperatorTransformer.scala

```diff
@@ -164,12 +164,15 @@ case class FilterExecTransformer(condition: Expression, child: SparkPlan)
     val operatorId = substraitContext.nextOperatorId
     // Firstly, need to check if the Substrait plan for this operator can be successfully generated.
     val relNode = try {
+      TransformerState.enterValidation
```

Comment:
Do we need to set validation state also for the other operators?

Reply:
I've thought about making a general mechanism, but didn't find any other scenarios.For example, doExecuteBroadcast for broadCastJoin, rddForSampling in RangePartitioner are not collected during validation.

Reply:
Thanks for your reply. I think we can continue to add it if we find other case. Just want to confirm, is any Subquery containing a Projection?

Reply:
Yes, subquery will appear in many places. Including filter, projection, aggregation expressions, etc.
Here is an example 
```sql
select (select avg(t_time_sk) from time_dim) as avg , (select max(t_time_sk) from time_dim) + (select min(t_time_sk) from time_dim) as range
```
After your reminder, i think i need to make some change in this PR to cover more scenarios.

Reply:
Thank you for that. I agree we can cover more common scenarios first in this PR.